src/stan/services/util/run_sampler.hpp

```diff
@@ -62,6 +62,9 @@ namespace stan {
         clock_t end = clock();
         double warm_delta_t = static_cast<double>(end - start) / CLOCKS_PER_SEC;
 
+        writer.write_adapt_finish(sampler);
```

Comment:
If `adapt_finish` is being called here to dump the non-adapted mass matrix then we should rename the function!